### PR TITLE
Implement push/pop context for AsyncIO

### PR DESCRIPTION
**Pull request by matan129**
In continuation of #268, I fixed some flaws in the original implementation.
Also, I fixed some other things along the way, hope it's not too messy.

Tests included.

### CODE REVIEW

File: tests/test_queues.py [hunk 1]
```diff
@@ -223,10 +223,10 @@ def test_redis_handler():
     import redis
     from logbook.queues import RedisHandler
 
-    KEY = 'redis'
+    KEY = 'redis-' + str(os.getpid())
```
Comment (matan129) [inline tests/test_queues.py]: I ran the tests with `tox -p auto`, which makes runs multiple test envs in parallel.
These changes make the tests not write to the same keys.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ services:
 - redis-server
 python:
 - '2.7'
+- '3.4'
```
Comment (vmalloc) [inline .travis.yml]: 3.4 is not a supported version of Logbook anymore, so please remove it from the matrix

File: README.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ For more information: http://logbook.readthedocs.org
 [ti]: https://secure.travis-ci.org/getlogbook/logbook.svg?branch=master
 [tl]: https://travis-ci.org/getlogbook/logbook
 [ai]: https://ci.appveyor.com/api/projects/status/quu99exa26e06npp?svg=true
-[vi]: https://img.shields.io/badge/python-2.6%2C2.7%2C3.3%2C3.4%2C3.5-green.svg
+[vi]: https://img.shields.io/badge/python-2.6%2C2.7%2C3.3%2C3.4%2C3.5%2C3.6%2C3.7-green.svg
```
Comment (vmalloc) [inline README.md]: 2.6 should be removed (not part of your change, but still while you're at it :-)

File: tests/test_queues.py (hunk 1 above)
Comment (vmalloc) [reply tests/test_queues.py]: Please change to `KEY = 'redis-{}'.format(os.getpid())`

File: tests/test_queues.py [hunk 4]
```diff
@@ -244,7 +244,7 @@ def test_redis_handler():
     assert message.find(LETTERS)
 
     # Change the key of the handler and check on redis
-    KEY = 'test_another_key'
+    KEY = 'test_another_key-' + str(os.getpid())
```
Comment (vmalloc) [inline tests/test_queues.py]: Same here

File: tests/test_queues.py [hunk 5]
```diff
@@ -293,7 +293,8 @@ def test_redis_handler_lpush():
     from logbook.queues import RedisHandler
     null_handler = logbook.NullHandler()
 
-    redis_handler = RedisHandler(key='lpushed', push_method='lpush',
+    KEY = 'lpushed-' + str(os.getpid())
```
Comment (vmalloc) [inline tests/test_queues.py]: Same here

**Comment by vmalloc** [comment]
Looks great! I haven't tried to run it with an actual asyncio code that attempts to manipulate contexts, but I trust that you have.
I added a few minor comments, and I'll merge it once it's done

**Comment by matan129** [comment]
I've fixed everything you requested.
I suggest that you squash-merge this PR, or if you want, I can do another interactive rebase and make the commits look nicer. (I didn't do it yet since it'd be easier to review)

**Comment by vmalloc** [comment]
I haven't forgotten about this - just having a rather busy week. Will take a closer look when I get the chance. Thanks!

**Comment by vmalloc** [comment]
Hi @matan129 Please rebase and squash commits while separating the "aesthetic" ones (like whitespaces/travis/badge changes from the actual contextvar support code.

Also if you can add a changelog entry for this change that would be great.

Thanks!

**Comment by matan129** [comment]
Done. I added the changelog entry under v1.5.0, I hope that's okay.

**Comment by vmalloc** [comment]
Great. Thanks a lot!

**Pull request merged**